src/DSUtil/text.cpp

```diff
@@ -142,6 +142,18 @@ CStringW UrlDecodeWithUTF8(CString in) {
     return in;
 }
 
+CStringW UrlGetPathname(CStringW in) {
+    int position = in.Find('?');
+    if (position > 0) return in.Left(position);
+    else if (position = 0) return _T("");

```

Comment:
Should be ==

A valid url never starts with ? or #, so the function can be optimized by removing the two "position==0" checks. Simply return unmodified string in that case. That is better than displaying empty string.

Reply:
I see.